realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java

```diff
@@ -160,15 +159,25 @@ public void onChange(AllJavaTypes object) {
         parent.setFieldObject(child);
         looperThreadRealm.commitTransaction();
 
-        assertEquals(2, looperThreadRealm.where(AllJavaTypes.class).findAll().size());
-        assertEquals(1, counter.get());
+        // This should be on the Looper queue, behind the notification
+        looperThread.postRunnable(
```

Comment:
It is smelly code, but if we haven't found a better way to solve it :man_shrugging: , and sorry :)

Every relevant listener should get notified on `beginTransaction()`, how many that is, probably depends since table-based listeners on RealmObjects are still a thing.

I cannot remember 100% how notifications are being enqueued for local transactions, I hope they are blocking but that needs to be verified. If they are not blocking, but use the notification worker thread they might have been enqueued when you call `postRunnable`.

My comment about async commits where perhaps a bit our of place, since, as you point out, there isn't any async commits here. I just wanted to point to the fact that they behave differently for future tests.